core/sdk-core/src/test/java/software/amazon/awssdk/core/internal/async/FileAsyncResponseTransformerTest.java

```diff
@@ -331,12 +332,25 @@ private static void stubException(String newContent, FileAsyncResponseTransforme
 
         RuntimeException runtimeException = new RuntimeException("oops");
         ByteBuffer content = ByteBuffer.wrap(newContent.getBytes(StandardCharsets.UTF_8));
-        transformer.onStream(SdkPublisher.adapt(Flowable.just(content, content)));
+        SdkPublisher<ByteBuffer> idlePublisher = new SdkPublisher<ByteBuffer>() {
+            @Override
+            public void subscribe(Subscriber<? super ByteBuffer> subscriber) {
+                subscriber.onSubscribe(new Subscription() {
```

Comment:
I'd rather use a `NoopSubscription` like we discussed. Here the content is still sent to the transformer by calling `subscriber.onNext()`, which might make the issue reappear.